src/Cache/FileCache.php

```diff
@@ -178,10 +179,34 @@ public function remove(string $key): bool
     {
         $file = $this->file($key);
 
-        if (is_file($file) === true) {
-            return F::remove($file);
-        } else {
-            return false;
+        if (is_file($file) === true && F::remove($file) === true) {
+            $this->removeEmptyDirectories(dirname($file));
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Removes empty directories safely by moving to the root directory.
+     *
+     * @param string $dir
+     * @return void
+     */
+    protected function removeEmptyDirectories(string $dir): void
+    {
+        try {
+            if (is_dir($dir) === true && $dir !== $this->root()) {
+                $files = array_diff(scandir($dir) ?? [], ['.', '..']);
+
+                if (empty($files) === true) {
+                    if (Dir::remove($dir) === true) {
+                        $this->removeEmptyDirectories(dirname($dir));
```

Comment:
The recursive approach will work quite well for shallow nesting levels (structures just a few levels deep). But if cache IDs contain potentially dozens of slashes (for whatever reason), this implementation will require too much memory for its purpose IMO.

We can fix this by using an iterative approach instead. Untested example code you could try:

```php
// ensure the path doesn't end with a slash for the next comparison
$dir = rtrim($dir, '/\/');

while (Str::startsWith($dir, $this->root()) === true && $dir !== $this->root()) {
    $files = array_diff(scandir($dir) ?? [], ['.', '..']);

    if (empty($files) === true && Dir::remove($dir) === true) {
        // continue with the next level up
        $dir = dirname($dir);
    }

    // no need to continue with the next level up as $dir was not deleted
    return;
}
```

Reply:
It was a 99% working code. Copy & paste & test 😅 Thank you 👍